test/e2e/apps/deployment.go

```diff
@@ -783,6 +783,10 @@ func testDeploymentsControllerRef(f *framework.Framework) {
 	framework.Logf("Waiting for the ReplicaSet to have the right controllerRef")
 	err = checkDeploymentReplicaSetsControllerRef(c, ns, deploy.UID, podLabels)
 	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying no extra ReplicaSet is created")
+	rsList := listAllReplicaSets(c, ns)
```

Comment:
Listing all ReplicaSets in the namespace happens to work because we create a namespace for each test, but I think it's more correct to only list ReplicaSets whose labels match, which you can do with `listDeploymentReplicaSets()`. Hypothetically, if there were other ReplicaSets in the namespace that don't match, it wouldn't mean this test failed.

Reply:
Ideally, we should also check that the one RS that exists has the same UID as the one RS we saw before orphaning, since the intention is to guarantee it will adopt that RS in particular.

Reply:
+1 on checking UID equality.